emsdk

```diff
@@ -1052,13 +1052,26 @@ def generate_dot_emscripten(active_tools):
   if embedded:
     cfg += "emsdk_path=os.path.dirname(os.environ.get('EM_CONFIG')).replace('\\\\', '/')\n"
 
+  # Different tools may provide the same activated configs; the latest to be
+  # activated is the relevant one.
+  final_activated_names = []
+  final_activated_values = {}
+
   for tool in active_tools:
     tool_cfg = tool.activated_config()
     if tool_cfg:
-      tool_cfg = tool_cfg.replace(';', '\n')
-      if 'SPIDERMONKEY_ENGINE=' in tool_cfg: has_spidermonkey = True
-      if 'NODE_JS=' in tool_cfg: has_node = True
-      cfg += tool_cfg + '\n'
+      for specific_cfg in tool_cfg.split(';'):
+        name, value = specific_cfg.split('=')
+        if name not in final_activated_values:
+          final_activated_names.append(name)
+        final_activated_values[name] = value
+
+  for name in final_activated_names:
+    if name == 'SPIDERMONKEY_ENGINE':
+      has_spidermonkey = True
+    if name == 'NODE_JS':
+      has_node = True
+    cfg += name + ' = ' + final_activated_values[name] + '\n'
```

Comment:
A bit surprised that this is needed, but looks good, certainly won't harm. It looks like the variable `final_activated_names` is being used explicitly for the purpose of preserving an ordered list, would be good to add a comment about that - or perhaps rename `final_activated_names` -> `activated_keys_in_order` and `final_activated_values` -> `activated_key_values` or similar?

Reply:
Sounds good, renamed as suggested.